src/app/screens/Send/index.tsx

```diff
@@ -1,29 +1,26 @@
-import {
-  CaretLeftIcon,
-  CrossIcon,
-  QrCodeIcon,
-} from "@bitcoin-design/bitcoin-icons-react/filled";
+import { CaretLeftIcon } from "@bitcoin-design/bitcoin-icons-react/filled";
 import Button from "@components/Button";
 import Container from "@components/Container";
 import Header from "@components/Header";
 import IconButton from "@components/IconButton";
-import QrcodeScanner from "@components/QrcodeScanner";
 import TextField from "@components/form/TextField";
 import lightningPayReq from "bolt11";
 import { useState } from "react";
 import { useTranslation } from "react-i18next";
-import { useNavigate } from "react-router-dom";
+import { useLocation, useNavigate } from "react-router-dom";
 import { toast } from "react-toastify";
+import QrcodeAdornment from "~/app/components/QrcodeAdornment";
+import { extractLightningTagData } from "~/app/utils";
 import lnurlLib from "~/common/lib/lnurl";
 import { isLNURLDetailsError } from "~/common/utils/typeHelpers";
 
 function Send() {
   const { t } = useTranslation("translation", { keyPrefix: "send" });
   const { t: tCommon } = useTranslation("common");
+  const location = useLocation();
 
-  const [invoice, setInvoice] = useState("");
+  const [invoice, setInvoice] = useState(location.state?.decodedText || "");
```

Comment:
Can you add a comment explaining why location.state is used? (comes from QRScanner screen). I wonder if decodedQR is better than decodedText?

Reply:
changed this to decodedQR. added comment for the use case of the location.state